doc/designs/functions-for-explicitly-fetched-signature-algorithms.md

```diff
@@ -0,0 +1,205 @@
+Functions for explicitly fetched PKEY algorithms
+================================================
+
+Quick background
+----------------
+
+There are several proposed designs that end up revolving around the same
+basic need, explicitly fetched signature algorithms.  The following method
+type is affected by this document:
+
+- `EVP_SIGNATURE`
+
+Public API - Add variants of `EVP_PKEY_CTX` functionality
+---------------------------------------------------------
+
+Through OTC discussions, it's been determined that the most suitable APIs to
+touch is the collection of `EVP_PKEY_` functions.  In this case, it involves
+`EVP_PKEY_sign()`, `EVP_PKEY_verify()`, `EVP_PKEY_verify_recover()` with
+associated functions.  They can be extended to accept an explicitly fetched
+algorithm of the right type, and to be able to process infinite amounts of
+data if the fetched algorithm permits it (typically, algorithms like ED25519
+or RSA-SHA256).
+
+It must be made clear that the added functionality can not be used to
+compose an algorithm from different parts.  For example, it's not possible
+to specify a `EVP_SIGNATURE` "RSA" and combine it with a parameter that
+specifies the hash "SHA256" to get the "RSA-SHA256" functionality.  For an
+`EVP_SIGNATURE` "RSA", the input is still expected to be a digest, or some
+other input that's limited to the modulus size of the RSA pkey.
```

Comment:
Editorial nit aside, I find this paragraph confusing. Please explain what you mean more clearly.
```suggestion
It must be made clear that the added functionality cannot be used to
compose an algorithm from different parts.  For example, it's not possible
to specify a `EVP_SIGNATURE` "RSA" and combine it with a parameter that
specifies the hash "SHA256" to get the "RSA-SHA256" functionality.  For an
`EVP_SIGNATURE` "RSA", the input is still expected to be a digest, or some
other input that's limited to the modulus size of the RSA pkey.
```